pkg/querier/queryrange/results_cache.go

```diff
@@ -55,6 +61,18 @@ func (e ExtractorFunc) Extract(start, end int64, from Response) Response {
 	return e(start, end, from)
 }
 
+// CacheSplitter generates cache keys
+type CacheSplitter interface {
```

Comment:
The introduction of this `CacheSplitter` looks an over complication to me. If we completely get rid of `SplitInterval`, is there any other reason to keep it?

Reply:
Because this is not implemented as a breaking change, we have multiple types which fulfill this interface and are chosen based on this criteria: https://github.com/cortexproject/cortex/pull/2040/files#diff-be5bce86bb962a94af90c800ee891d85R97-R109

Furthermore it's a convenient place to have an interface for downstream consumers. I know we'd like to parameterize this based on tenant overrides in loki, for example.